File: app/src/main/java/de/tum/in/tumcampusapp/models/tumcabe/News.java
package de.tum.in.tumcampusapp.models.tumcabe;

import java.util.Date;

import de.tum.in.tumcampusapp.auxiliary.Utils;

/**
 * News Object
 */
public class News {

    /**
     * News Facebook-ID
     */
    public final String id;

    /**
     * Local image, e.g. /mnt/sdcard/tumcampus/news/cache/xy.jpg
     */
    public final String image;

    /**
     * Link Url, e.g. http://www.in.tum.de
     */
    public final String link;

    /**
     * Content
     */
    public final String title;

    /**
     * Date
     */
    public final Date date;
    public final Date created;

    public final String src;

    /**
     * New News
     *
     * @param id      News Facebook-ID
     * @param title   Title
     * @param link    Url, e.g. http://www.in.tum.de
     * @param image   Image url e.g. http://www.tu-film.de/img/film/poster/Fack%20ju%20Ghte.jpg
     * @param date    Date
     * @param created Creation date
     */
    public News(String id, String title, String link, String src, String image, Date date, Date created) {
        this.id = id;
        this.title = title;
        this.link = link;
        this.src = src;
        this.image = image;
        this.date = date;
        this.created = created;
    }

    @Override
    public String toString() {
        return "id=" + id + " title=" + title + " link=" + link + " image="
               + image + " date=" + Utils.getDateString(date) + " created=" + Utils.getDateString(created);
    }
}

File: app/src/main/java/de/tum/in/tumcampusapp/models/tumcabe/News.kt
package de.tum.`in`.tumcampusapp.models.tumcabe

import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey
import java.util.*

/**
 * New News
 *
 * @param id      News Facebook-ID
 * @param title   Title
 * @param link    Url, e.g. http://www.in.tum.de
 * @param image   Image url e.g. http://www.tu-film.de/img/film/poster/Fack%20ju%20Ghte.jpg
 * @param date    Date
 * @param created Creation date
 */
@Entity
data class News(@PrimaryKey
                var id: String = "",
                var title: String = "",
                var link: String = "",
                var src: String = "",
                var image: String = "",
                var date: Date = Date(),
                var created: Date = Date(),
                var dismissed: Int = 0) {
    /**
     * Identifies News as a film.
     *
     * @return true if News is a film; else false
     */
    fun isFilm(): Boolean {
        return src == "2"
    }

}